### ISSUE TITLE
Use an existing table for Chartist data

### ISSUE THREAD

**Issue by Founts** [chartist-js/chartist#150]
If you have a table element full of data on a page, it would be awesome to add a Chartist visualization based on that table and call it a day.  A table adds some accessibility to the page, gives an alternate detailed view of information and is a fallback should something go wrong with JS (highly unlikely but hey). 

Per Gion, I'm opening this issue for discussion and solutions. Thoughts?

**Comment by gionkunz** [comment]
I'd appreciate that we build a Chartist plugin for this functionality. http://gionkunz.github.io/chartist-js/plugins.html

Plugins work like custom functionality added via Event interception. Currently there is no event that allows you to intercept the process where the data (series values and labels) is getting processed but I've planned to add this event also for solving some data transformation requirements using event interception and plugins (e.g. #139).

Once we have the data transformation interception event this will allow us to create a plugin that can be used to specify a selector for a table on the page and the plugin will gather the data from the table instead of using the original data. This should do the job nicely and we don't need to bloat the core functionality of Chartist.js.

I'm open for any ideas or suggestions.

**Comment by Founts** [comment]
@gionkunz Sounds ideal to me.  If I had the skills to hack away at this quickly I would. I'm not there. This may help though: https://github.com/bbgvisualjournalist/table-to-svg

**Comment by Founts** [comment]
@gionkunz another route to consider (or maybe another offering) - Google Charts generates an accessible table based on the data you tell it to map.  I think this is the functionality: https://developers.google.com/chart/interactive/docs/reference#DataView
One reason to go down this road is Chartist would generate the table and it therefore would be well formed (provided the data you provided to Chartist is good) whereas reading from a table that has garbage means you might have to trap for the garbage in code. Just some more thoughts...

**Comment by gionkunz** [comment]
Yeah, I could see two modes of the plugin. One would be that it generates a table that is visually hidden and the other one is that it grabs an existing table and uses it to generate the chart (and maybe optionally visually hide the table).

**Comment by gionkunz** [comment]
@JamieKnight maybe you have some input here too? Also if we use a plugin that generates an accessibility table, should we set the whole chart element to aria-hidden true?
